### PR TITLE
Add FVM config

### PR DESCRIPTION
**Pull request by easazade**
Adding [fvm](https://fvm.app/) config for flutter version management to set the flutter version used for development. this will help contributors to know and use the same flutter version when developing and running pub commands

### CODE REVIEW

**Comment by easazade** [comment]
also fixed a overflow error that was showing in example app
Screenshot 2024-06-13 at 16 18 02

**Comment by luis901101** [comment]
Hi @easazade thanks for your contribution

**Comment by luis901101** [comment]
> Adding [fvm](https://fvm.app/) config for flutter version management to set the flutter version used for development. this will help contributors to know and use the same flutter version when developing and running pub commands

Regarding the `fvm` config, I'm not agree with adding this to the project repo, I mean anyone should be able to collaborate using whatever tools or setup they decide, also for the required version the `pubspec.yaml` is the one to define that, like:

```yaml
environment:
  sdk: '>=3.0.0 <4.0.0'
  flutter: '>=3.3.0'
```

Maybe the minimum flutter version could be increased, but as far as I remember the package didn't require a greater version so for compatibility purposes I leave it as 3.3.0.

File: .vscode/settings.json
Comment (luis901101) [inline .vscode/settings.json:1]: IDE settings should not be included in the repo

File: example/lib/main.dart [hunk 1]
```diff
@@ -21,19 +21,20 @@ class MyApp extends StatelessWidget {
...
+          title: 'FHIR Questionnaire Demo',
+          scrollBehavior: const CustomScrollBehavior(),
+          theme: ThemeData(
+              colorScheme: ColorScheme.fromSeed(seedColor: Colors.green),
+              useMaterial3: true,
+              inputDecorationTheme: snapshot.data),
+          home: const MyHomePage(),
+        );
+      },
+    );
```
Comment (luis901101) [inline example/lib/main.dart:37]: This seems like a change in the format, run a `dart format .` to apply the default format.

File: example/lib/main.dart [hunk 2]
```diff
@@ -157,7 +158,10 @@ class _MyHomePageState extends State<MyHomePage> {
                   items: inputDecorationThemes
                       .map((e) => DropdownMenuItem<InputDecorationTheme>(
                             value: e.value,
-                            child: Text(e.name),
+                            child: Text(
+                              e.name,
+                              style: const TextStyle(fontSize: 14),
+                            ),
```
Comment (luis901101) [inline example/lib/main.dart:164]: Here the base issue is the length of the text of one of the items which is too long: `"Outline Streched Rounded Filled No Borders"`, reducing the font size will only work if the text length remains the same and the screen density is not reduced like in older devices... maybe a better approach would be to limit the width of the Text container so it never overlaps with the drop down arrow button, something like:

```dart
DropdownMenuItem<InputDecorationTheme>(
  value: e.value,
  child: SizedBox(
    width: MediaQuery.of(context).size.width * 0.82,
    child: Text(
      e.name,
    ),
  ),
)
```
Anyway your approach is valid as well as this is the sample project.

File: .vscode/settings.json
Comment (easazade) [reply .vscode/settings.json:1]: About the fvm config. I can remove it entirely. or just change it to not enforce flutter version on contributors. for the latter option i should remove IDE settings config and just keep `.fvmrc` file. this way anyone with any flutter setup can contribute. what do you think?

File: example/lib/main.dart (hunk 1 above)
Comment (easazade) [reply example/lib/main.dart:37]: I added a `,` betweek `})`. which changed the style after format. do you want to revert it?

File: example/lib/main.dart (hunk 2 above)
Comment (easazade) [reply example/lib/main.dart:164]: this is better

**Comment by easazade** [comment]
@luis901101 

just a question not regarding this PR: 
in library code there is QuestionnaireUtils that has some sample questionnaires that are only used in example app. should it be there instead?

File: .fvmrc
Comment (luis901101) [inline .fvmrc:1]: This file is still included, maybe requires a `git rm --cache .fvmrc`

**Comment by luis901101** [comment]
> @luis901101
> 
> just a question not regarding this PR: in library code there is QuestionnaireUtils class that has some sample questionnaires that are only used in example app. should it be there instead?

Yeah, it should not be in the QuestionnaireUtils it makes more sense to be a utils class only for the example project, if you want you can change it and commit.

File: .fvmrc
Comment (easazade) [reply .fvmrc:1]: [174c6a0](https://github.com/luis901101/fhir_questionnaire/pull/1/commits/174c6a07a932d77b21122fef54c2126ffcc28413)

**Pull request merged**